### PR TITLE
Upgrade helm command to support only helm 3

### PR DESCRIPTION
**Pull request by jiacheliu3**
This addresses https://github.com/Alluxio/alluxio/issues/11319

We upgrade the helm template generation commands from Helm 2 to Helm 3.

### CODE REVIEW

File: integration/kubernetes/multiMaster-embeddedJournal/master/alluxio-master-service.yaml.template [hunk 1]
```diff
@@ -41,17 +29,30 @@ spec:
...
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+
```
Comment (jiacheliu3) [inline integration/kubernetes/multiMaster-embeddedJournal/master/alluxio-master-service.yaml.template:46]: Seems Helm 3 has its own understanding of where the comments should appear in a range loop. This is really hilarious. @madanadit

File: integration/kubernetes/multiMaster-embeddedJournal/master/alluxio-master-service.yaml.template [hunk 2]
```diff
@@ -72,17 +73,18 @@ spec:
     role: alluxio-master
     app: alluxio
     release: alluxio
-    statefulset.kubernetes.io/pod-name: alluxio-master-1
+    statefulset.kubernetes.io/pod-name: alluxio-master-0
 ---
+# Source: alluxio/templates/master/service.yaml
```
Comment (jiacheliu3) [inline integration/kubernetes/multiMaster-embeddedJournal/master/alluxio-master-service.yaml.template:78]: Seems they are generated in order but output separately. @madanadit

File: integration/kubernetes/multiMaster-embeddedJournal/master/alluxio-master-service.yaml.template [hunk 3]
```diff
@@ -103,6 +105,4 @@ spec:
     role: alluxio-master
     app: alluxio
     release: alluxio
-    statefulset.kubernetes.io/pod-name: alluxio-master-2
----
-
```
Comment (jiacheliu3) [inline integration/kubernetes/multiMaster-embeddedJournal/master/alluxio-master-service.yaml.template:108]: Also the output are all trimmed and lost the ending empty row. That's fine to me as we are removing these from the code repo.

**Comment by jiacheliu3** [comment]
@madanadit PTAL at the change we need to upgrade to Helm 3. A non-trivial rebase is needed after https://github.com/Alluxio/alluxio/pull/11321 is in. 

Also the change in docs is non-trivial. I may address that in a separate PR depending on the effort needed.

**Comment by jiacheliu3** [comment]
> @madanadit PTAL at the change we need to upgrade to Helm 3. A non-trivial rebase is needed after #11321 is in.
> 
> Also the change in docs is non-trivial. I may address that in a separate PR depending on the effort needed.

The doc change is in this PR. Also the rebase is done. @madanadit PTAL. Thanks!

**Comment by jiacheliu3** [comment]
```
Tests in error: 
alluxio.server.ft.journal.TriggeredCheckpointTest.ufsJournal(alluxio.server.ft.journal.TriggeredCheckpointTest)
  Run 1: TriggeredCheckpointTest.ufsJournal:63->validateCheckpointInClusterRestart:124 » Timeout
  Run 2: TriggeredCheckpointTest.ufsJournal:63->validateCheckpointInClusterRestart:124 » Timeout
```

**Comment by jiacheliu3** [comment]
alluxio-bot, test this please

**Comment by jiacheliu3** [comment]
@madanadit Thanks! The PR is rebased and good to go.

**Comment by madanadit** [comment]
alluxio-bot, merge this please.

**Pull request merged**